extensions/indexes/lucene/src/main/java/org/exist/indexing/lucene/LuceneFieldConfig.java

```diff
@@ -160,7 +171,12 @@ private boolean checkCondition(DBBroker broker, DocumentImpl document, NodeId no
     protected void processResult(Sequence result, Document luceneDoc) throws XPathException {
         for (SequenceIterator i = result.unorderedIterator(); i.hasNext(); ) {
             final String text = i.nextItem().getStringValue();
-            final Field field = convertToField(text);
+            final Field field;
```

Comment:
final Field field = binary ? convertToDocValue(text) : convertToField(text);

would be acceptable for me :-)

Reply:
for me as well, but I am totally indifferent on this
if-else is deemed to be easier to understand